python/ray/rllib/utils/result.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from ray.tune.result import TrainingResult
+
+
+class RLTrainingResult(TrainingResult):
+    """TrainingResult for RLlib.
+
+    In RLlib, the supplied algorithms fill in TrainingResult for you.
+
+    Attributes:
+        timesteps_total (int): (Required) Accumulated timesteps for this
+            entire experiment.
+        timesteps_this_iter (int): (Auto-filled) Number of timesteps
+            in the simulator in this iteration.
+        info (dict): (Optional) Custom metadata to report for this iteration.
+        episode_reward_mean (float): (Optional) The mean episode reward
+            if applicable.
+        episode_reward_min (float): (Optional) The min episode reward
+            if applicable.
+        episode_reward_max (float): (Optional) The max episode reward
+            if applicable.
+        episode_len_mean (float): (Optional) The mean episode length
+            if applicable.
+        episodes_total (int): (Optional) The number of episodes total.
+        policy_reward_mean (float): (Optional) Per-policy reward information
+            in multi-agent RL.
+        mean_accuracy (float): (Optional) The current training accuracy
+            if applicable.
+        mean_validation_accuracy (float): (Optional) The current
+            validation accuracy if applicable.
+        mean_loss (float): (Optional) The current training loss if applicable.
+        neg_mean_loss (float): (Auto-filled) The negated current training loss.
+    """
+
+    @property
```

Comment:
This is somewhat overkill, you can just do
```
def __getattr__(self, key):
    return self[key]
```

and raise the right ValueError otherwise.